ext/liquid_c/tokenizer.c

```diff
@@ -66,20 +67,29 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
 
     token->str = cursor;
     token->type = TOKEN_RAW;
+    token->lstrip = 0;
+    token->rstrip = 0;
 
     while (cursor < last) {
         if (*cursor++ != '{')
             continue;
 
         char c = *cursor++;
+        if (cursor <= last && *cursor == '-') {
+          cursor++;
+          token->rstrip = 1;
+        }
         if (c != '%' && c != '{')
             continue;
-        if (cursor - tokenizer->cursor > 2) {
+        if (cursor - tokenizer->cursor > 2 + token->rstrip) {
             token->type = TOKEN_RAW;
-            cursor -= 2;
+            cursor -= 2 + token->rstrip;
+            token->lstrip = tokenizer->lstrip_flag;
+            tokenizer->lstrip_flag = 0;
             goto found;
         }
         token->type = TOKEN_INVALID;
+        token->lstrip = token->rstrip;
```

Comment:
This will leave `token->rstrip` set to 1, even though it is just the left whitespace control character that has been found.  Below you only conditionally set `token->rstrip = 1`, so left whitespace control character will always result in the rstrip flag being set on the token.